### ISSUE TITLE
Suddenly seeing random timeouts across more than one service (DynamoDB, S3)

### ISSUE THREAD

**Issue by alexpaluzzi** [aws/aws-sdk-js#3609]
Confirm by changing [ ] to [x] below:
- [x] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/welcome.html) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js) for answers

Having the latest v2 version of the SDK running locally with very simple DynamoDB queries and S3 GetObjects in large loops working great for the last 2 weeks. Suddenly starting Saturday morning, I am seeing periodic timeouts:

```
Error: read ETIMEDOUT
    at TLSWrap.onStreamRead (internal/stream_base_commons.js:209:20) {
  errno: -60,
  code: 'ETIMEDOUT',
  syscall: 'read',
  name: 'TimeoutError',
  '$metadata': { attempts: 1, totalRetryDelay: 0 }
}
```

I was convinced it was some kind of outage but I haven't seen anything. I have tried multiple computers, operating systems and internet connections. I then upgraded the code to use the v3 DynamoDB client and I am seeing the exact same issue.

It seems to be fine if I pull 1-2 in a quick loop. But if I do more than that, there's almost always at least one that hangs for 30 seconds and retries again. I can run 600 Dynamo queries in a loop (in parallel) and 595 will work fine. Or I can run 4 in the same loop and 3 will work fine. I have tried every combination of http agent options and `keepAlive`.

I realize it’s not this library if it was working before without code changes but I’m hoping there’s some guidance on how I can find the underlying cause or if there’s an option or configuration somewhere.

**Comment by alexpaluzzi** [comment]
Whelp, just about 48 hours after it started happening, it has resolved itself. I can only assume there was an issue on Amazon's end this weekend in my region or something like that. Sorry for the false alarm. Cheers.

**Comment by arifali123** [comment]
I got the same issue my s3 will not work, I get the dumbest errors. Like I upload a file to s3 and it uploads file all the bytes are sent and it shows up on the web but on the sdk keeps returning timeout error. Insanely annoying please fix this amazon.

**Comment by ajredniwja** [comment]
@arifali123, can you please open a new issue with steps to reproduce your case if possible.
